crates/wasmtime/src/runtime/component/values.rs

```diff
@@ -198,9 +202,9 @@ impl Val {
 
                 Val::Flags(flags.into())
             }
-            InterfaceType::Future(_)
-            | InterfaceType::Stream(_)
-            | InterfaceType::ErrorContext(_) => todo!(),
+            InterfaceType::Future(_) => FutureReader::<()>::lift(cx, ty, next(src))?.into_val(),
```

Comment:
The `::<()>` bit here and below gives me a bit of pause. Is that temporary or final-into-the-future?

Reply:
It's "final" in that I wasn't planning to change it, but I agree it looks weird.  I had already written the static API `Lift`/`Lower` impls for these types, so I figured I'd reuse those implementations here since they're payload-type-agnostic.  I'll move the lifting and lowering code into separate, top-level functions which both the static API and dynamic API can use, which will avoid this.